mono/mini/aot-compiler.c

```diff
@@ -8327,7 +8327,17 @@ emit_llvm_file (MonoAotCompile *acfg)
 	if (acfg->aot_opts.llvm_only) {
 		/* Use the stock clang from xcode */
 		// FIXME: arch
-		command = g_strdup_printf ("clang++ -fexceptions -march=x86-64 -fpic -msse -msse2 -msse3 -msse4 -O2 -fno-optimize-sibling-calls -Wno-override-module -c -o \"%s\" \"%s.opt.bc\"", acfg->llvm_ofile, acfg->tmpbasename);
+                command = NULL;
+#if TARGET_ARM
+		command = g_strdup_printf ("clang++ -fexceptions -march=armv7 -fpic  -O2 -fno-optimize-sibling-calls -Wno-override-module -c -o \"%s\" \"%s.opt.bc\"", acfg->llvm_ofile, acfg->tmpbasename);
+#endif
+
+#ifdef TARGET_AMD64
+		command = g_strdup_printf ("clang++ -fexceptions -march=x86_64 -fpic -msse -msse2 -msse3 -msse4 -O2 -fno-optimize-sibling-calls -Wno-override-module -c -o \"%s\" \"%s.opt.bc\"", acfg->llvm_ofile, acfg->tmpbasename);
+#endif
```

Comment:
Looks like `x86_64` is the right one, based on `parseArch` in `lib/Support/Triple.cpp` in LLVM.